File: leetcode/674-longest-continuous-increasing-subsequence/longest-continuous-increasing-subsequence.cpp
class Solution {
public:
    int findLengthOfLCIS(vector<int>& nums) {
        int count = 1;
        int maxCount = 1;
        int n = nums.size();

        for (int i = 0; i < n - 1; i++) {
            if (nums[i] < nums[i+1]) {
                count++;
                maxCount = max(maxCount, count);
            }
            else {
                count = 1;
            }
        }
        return maxCount;
    }
};

File: leetcode/674-longest-continuous-increasing-subsequence/longest-continuous-increasing-subsequence.py
class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        ans, max_ans = 1, 1

        for i in range(1,len(nums)):
            if nums[i] > nums[i-1]:
                ans += 1 
            else:
                max_ans = max(max_ans, ans)
                ans = 1

        return max(ans, max_ans)
        